Userland/Libraries/LibWeb/Layout/GridFormattingContext.h

```diff
@@ -22,6 +22,26 @@ class GridFormattingContext final : public BlockFormattingContext {
 
 private:
     float m_automatic_content_height { 0 };
+    bool is_auto_positioned_row(CSS::GridTrackPlacement const&, CSS::GridTrackPlacement const&) const;
+    bool is_auto_positioned_column(CSS::GridTrackPlacement const&, CSS::GridTrackPlacement const&) const;
+    bool is_auto_positioned_track(CSS::GridTrackPlacement const&, CSS::GridTrackPlacement const&) const;
+};
+
+class OccupationGrid {
+public:
+    OccupationGrid(int, int);
+
+    void maybe_add_column(int);
+    void maybe_add_row(int);
+    void set_occupied(int, int, int, int);
+    void set_occupied(int, int);
+
+    int column_count() { return static_cast<int>(m_occupation_grid[0].size()); }
+    int row_count() { return static_cast<int>(m_occupation_grid.size()); }
+    bool is_occupied(int, int);
```

Comment:
Ninja nit: Don't omit the names of parameters in header files unless it's obvious what they are.

An example of when you could omit it is if the type were `Gfx::Painter` where it's obviously a painter, but an `int` could be anything.

Reply:
Ok! Makes sense